crates/sui-types/src/sui_system_state.rs

```diff
@@ -44,6 +48,17 @@ pub struct ValidatorMetadata {
     pub next_epoch_gas_price: u64,
 }
 
+impl ValidatorMetadata {
+    pub fn to_validator_and_stake_pair(&self) -> (AuthorityName, StakeUnit) {
+        (
+            // TODO: Make sure we are actually verifying this on-chain.
```

Comment:
what's involved in addressing this TODO?

Reply:
When a transaction is made to request adding a new validator, in the move contract we need to make sure that the information provided is legal.